src/TempDirectory.cc

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <ignition/common/TempDirectory.hh>
+#include <ignition/common/Console.hh>
+
+#ifdef _WIN32
+#include <windows.h>
+#include <direct.h>
+#include <fileapi.h>
+#include <io.h>
+#endif
+
+using namespace ignition;
+using namespace common;
+
+/////////////////////////////////////////////////
+// Return true if success, false if error
+inline bool fs_warn(const std::string &_fcn,
+             const std::error_code &_ec,
+             const FilesystemWarningOp &_warningOp = FSWO_LOG_WARNINGS)
+{
+
+  if (_ec)
+  {
+    if (FSWO_LOG_WARNINGS == _warningOp)
+    {
+      ignwarn << "Failed ignition::common::" << _fcn
+        << " (ec: " << _ec << " " << _ec.message() << ")\n";
+    }
+    return false;
+  }
+  return true;
+}
+
+
+/////////////////////////////////////////////////
+std::string temp_directory_path(std::error_code& _err)
```

Comment:
Style nit: 
```suggestion
std::string tempDirectoryPath(std::error_code& _err)
```

There are also instances within the function

Reply:
This is a local implementation of std::filesystem::temp_directory_path.  It is not part of the public API and can be removed as soon as std::filesystem is supported everywhere necessary.  I have noted this in a comment.

Reply:
Here is where it is removed in the switch to std::filesystem:

https://github.com/ignitionrobotics/ign-common/pull/238/files#diff-cc8a2d7ec160eade949b51ef4324d5784f463d271eea0c3fd23b52e14e057a74L50-L74